include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -120,7 +120,7 @@ class ValueLoc
   constexpr ValueLoc() = default;
   constexpr ValueLoc(ValueLoc const &) = default;
 
-  RAJA_HOST_DEVICE ValueLoc &operator=(ValueLoc const &) = default;
```

Comment:
RAJA_HOST_DEVICE here is necessary for SoAArray/Ptr in the CUDA reducer implementation. We probably need to leave this in and live with the warnings. Looping in @MrBurmark .

Reply:
Yes, I will revert this. See my comment above.

Reply:
Sounds good. We need to decide on CUDA version compatibility. If we are going to move up to only supporting 10+, I can update the Travis images.

Reply:
Well, we need to discuss this. I've been testing some things on Argonne machines for ATPESC and they have really old compilers.